src/Build/BackEnd/Components/SdkResolution/CachingSdkResolverService.cs

```diff
@@ -57,7 +55,21 @@ public override SdkResult ResolveSdk(int submissionId, SdkReference sdk, Logging
                  */
                 Lazy<SdkResult> resultLazy = cached.GetOrAdd(
                     sdk.Name,
-                    key => new Lazy<SdkResult>(() => base.ResolveSdk(submissionId, sdk, loggingContext, sdkReferenceLocation, solutionPath, projectPath, interactive, isRunningInVisualStudio)));
+                    key => new Lazy<SdkResult>(() =>
+                    {
+                        MSBuildEventSource.Log.CachedSdkResolverServiceResolveSdkStart(sdk.Name, solutionPath, projectPath);
+
+                        SdkResult result = base.ResolveSdk(submissionId, sdk, loggingContext, sdkReferenceLocation, solutionPath, projectPath, interactive, isRunningInVisualStudio);
+
+                        MSBuildEventSource.Log.CachedSdkResolverServiceResolveSdkStop(sdk.Name, solutionPath, projectPath, result.Success);
+
+                        return result;
+                    }));
+
+                if (resultLazy.IsValueCreated)
+                {
+                    MSBuildEventSource.Log.CachedSdkResolverServiceResolveSdkFromCache(sdk.Name, solutionPath, projectPath, resultLazy.Value.Success);
```

Comment:
This means you'll get a single event showing that it was resolved from the cache (if it was), but it won't give you how long that took. Maybe an event before and after where the after also includes resultLazy.IsValueCreated as a parameter?

Reply:
Technically getting a cached result should be nearly instantaneous, as we're only waiting for a lock.  I thought it didn't make sense to measure that, but am open to changing it.

Reply:
Right, but I don't think it's bad to verify that. More importantly, for not cached results, it may be useful to know how long it took, and we'd lose that functionality with its current state.

Reply:
Okay I've removed the "FromCache" events and added a boolean to the start and stop events to indicate whether or not the result was cached.  Much cleaner, thanks for the suggestion!